packages/valory/skills/decision_maker_abci/behaviours/reedem.py

```diff
@@ -0,0 +1,375 @@
+            redeemer=self.safe_address_lower,
+            collateral_token=self.current_collateral_token,
+            parent_collection_id=ZERO_BYTES,
+            condition_id=self.current_condition_id,
+            index_sets=self.current_index_sets,
+        )
+        return result
+
+    def _check_already_resolved(self) -> WaitableConditionType:
+        """Check whether someone has already resolved for this market."""
+        result = yield from self._conditional_tokens_interact(
+            performative=ContractApiMessage.Performative.GET_STATE,
+            contract_callable=get_name(ConditionalTokensContract.check_resolved),
+            data_key="resolved",
+            placeholder=get_name(RedeemBehaviour.waitable_flag_result),
+            condition_id=self.current_condition_id,
+        )
+        return result
+
+    def _build_resolve_data(self) -> WaitableConditionType:
+        """Prepare the safe tx to resolve the condition."""
+        result = yield from self.contract_interact(
+            performative=ContractApiMessage.Performative.GET_STATE,  # type: ignore
+            contract_address=self.params.realitio_proxy_address,
+            contract_public_id=RealitioProxyContract.contract_id,
+            contract_callable=get_name(RealitioProxyContract.build_resolve_tx),
+            data_key="data",
+            placeholder=get_name(RedeemBehaviour.built_data),
+            question_id=self.current_question_id,
+            template_id=self.current_fpmm.templateId,
+            question=self.current_fpmm.question.data,
+            num_outcomes=self.current_condition.outcomeSlotCount,
+        )
+
+        if not result:
+            return False
+
+        batch = MultisendBatch(
+            to=self.params.realitio_proxy_address,
+            data=HexBytes(self.built_data),
+        )
+        self.multisend_batches.append(batch)
+        return True
+
+    def _build_claim_data(self) -> WaitableConditionType:
+        """Prepare the safe tx to claim the winnings."""
+        answer_data = self.current_fpmm.question.answer_data
+        result = yield from self.contract_interact(
+            performative=ContractApiMessage.Performative.GET_STATE,  # type: ignore
+            contract_address=self.params.realitio_address,
+            contract_public_id=RealitioContract.contract_id,
+            contract_callable=get_name(RealitioContract.build_claim_winnings),
+            data_key="data",
+            placeholder=get_name(RedeemBehaviour.built_data),
+            question_id=self.current_question_id,
+            # `history_hashes` is calculated as described here:
+            # https://realitio.github.io/docs/html/contract_explanation.html#answer-history-entries.
+            # however, the current implementation of the service does not place multiple answers for a single market.
+            # therefore, this value is always set to zero bytes
+            history_hashes=ZERO_BYTES,
```

Comment:
i believe this needs to be an array, no?

Reply:
Well detected, thanks!